source/common/grpc/async_client_impl.h

```diff
@@ -158,12 +158,13 @@ class AsyncStreamImpl : public AsyncStream<RequestType>,
       streamError(Status::GrpcStatus::Internal);
       return;
     }
+    std::string grpc_message = Common::getGrpcMessage(*trailers);
     if (grpc_status.value() != Status::GrpcStatus::Ok) {
-      streamError(grpc_status.value());
+      streamError(grpc_status.value(), std::move(grpc_message));
```

Comment:
nit: std::move here doesn't actually do anything since you are passing const reference. If you want to make this more efficient, I would actually return a `const HeaderEntry*` from `getGrpcMessage` and pass that around. I don't feel that strong though.